includes/admin/class-wc-rest-stripe-settings-controller.php

```diff
@@ -58,62 +56,151 @@ public function register_routes() {
-							'enum' => array_keys( $form_fields['payment_request_button_theme']['options'] ),
-						],
+						// it can happen that `$form_fields['payment_request_button_size']` is empty (in tests) - fixing temporarily.
+						'enum'              => array_keys( isset( $form_fields['payment_request_button_size']['options'] ) ? $form_fields['payment_request_button_size']['options'] : [] ),
 						'validate_callback' => 'rest_validate_request_arg',
 					],
-					'enabled_payment_method_ids'       => [
-						'description'       => __( 'Payment method IDs that should be enabled. Other methods will be disabled.', 'woocommerce-gateway-stripe' ),
+					'payment_request_button_locations'      => [
+						'description'       => __( 'Express checkout locations that should be enabled.', 'woocommerce-gateway-stripe' ),
 						'type'              => 'array',
 						'items'             => [
 							'type' => 'string',
-							'enum' => $this->gateway->get_upe_available_payment_methods(),
+							'enum' => array_keys( $form_fields['payment_request_button_locations']['options'] ),
 						],
 						'validate_callback' => 'rest_validate_request_arg',
 					],
+					'is_manual_capture_enabled'             => [
+						'description'       => __( 'If manual capture of charges should be enabled.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'boolean',
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+					'is_saved_cards_enabled'                => [
+						'description'       => __( 'If "Saved cards" should be enabled.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'boolean',
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+					'is_separate_card_form_enabled'         => [
+						'description'       => __( 'If credit card number field, expiry date field, and CVC field should be separate.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'boolean',
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+					'statement_descriptor'                  => [
+						'description'       => __( 'Bank account descriptor to be displayed in customers\' bank accounts.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'string',
+						'validate_callback' => [ $this, 'validate_regular_statement_descriptor' ],
+					],
+					'is_short_statement_descriptor_enabled' => [
+						'description'       => __( 'When enabled, we\'ll include the order number for card and express checkout transactions.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'boolean',
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+					'short_statement_descriptor'            => [
+						'description'       => __( 'We\'ll use the short version in combination with the customer order number.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'string',
+						'validate_callback' => [ $this, 'validate_short_statement_descriptor' ],
+					],
```

Comment:
I'm getting a validation error on this.
When I _don't_ provide a "short" statement descriptor and the "Add customer order number to the bank statement" checkbox is un-checkeched, the validation fails.
Should it be failing, since there is no short statement descriptor?
![Screen Shot 2021-10-06 at 2 15 31 PM](https://user-images.githubusercontent.com/273592/136268340-184ded6c-f385-4f17-b425-79ac879eb57b.png)